src/Functions/MultiMatchAllIndicesImpl.h

```diff
@@ -102,11 +103,6 @@ struct MultiMatchAllIndicesImpl
             offset = haystack_offsets[i];
         }
 #else
-        (void)haystack_data;
```

Comment:
I'm not sure that replacing ```(void)arg``` with  ```[[maybe_unused]]  arg``` is good in this particular case.
Previously unused args where silenced only inside ```#else```  branch and now they are silenced both in ```#if``` and ```#else``` branches